### PR TITLE
feat: Added support for custom tags on ec2 deploys

### PR DESCRIPTION
**Pull request by jvasallo**
Initial commit getting custom tags to work in deploy stage. Need to continue and add support for other deploy types and custom infra.

- [x] EC2
- [x] ELB
- [x] Lambda
- [x] S3
- [x] etc

### CODE REVIEW

**Comment by jvasallo** [comment]
ELB in Spinnaker currently does not support tagging but can be added later when fixed. 

Once tests pass this should be good.

File: src/foremast/templates/pipeline/stage-deploy.json.j2 [hunk 1]
```diff
@@ -35,6 +35,11 @@
           "suspendedProcesses": [],
           "securityGroups": {{ data.app.instance_security_groups }},
           "tags": {
+            {% if data.app.custom_tags %}
```
Comment (linjmeyer) [inline src/foremast/templates/pipeline/stage-deploy.json.j2:38]: I think tags is missing the closing bracket and comma, and the loop creates a comma on the last item.  Maybe `tags: {{ data.app.custom_tags | tojson }},` would be easier?

File: src/foremast/templates/configs/configs.json.j2 [hunk 2]
```diff
@@ -5,6 +5,7 @@
         "approval_timeout": null,
         "archaius_enabled": false,
         "canary": false,
+        "custom_tags": {},
```
Comment (linjmeyer) [inline src/foremast/templates/configs/configs.json.j2:8]: One way to add tags for all the AWS pipeline types is great 👍

**Review by linjmeyer — CHANGES_REQUESTED** [review]

File: src/foremast/templates/configs/configs.json.j2 (hunk 2 above)
Comment (jvasallo) [reply src/foremast/templates/configs/configs.json.j2:8]: Could extend to GCP too if ever needed. :)

File: src/foremast/templates/pipeline/stage-deploy.json.j2 (hunk 1 above)
Comment (jvasallo) [reply src/foremast/templates/pipeline/stage-deploy.json.j2:38]: @linjmeyer We traditionally always tagged `app_group`, `app_name`, and `owner_email` as you see below. 👍 

We could change it `tojson` however, we might need to refactor to include it in the ec2 code.

**Pull request merged**